### PR TITLE
haproxy addons

### PR DESCRIPTION
**Pull request by cosmosified**
these aren't clean. but if anyone wants to at least use them they work

### CODE REVIEW

File: plugins/modules/pfsense_haproxy_frontend.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: pfsense_haproxy_frontend
+version_added: "2.10"
```
Comment (opoplawski) [inline plugins/modules/pfsense_haproxy_frontend.py:17]: This should be 0.6.0 at the moment.

File: plugins/modules/pfsense_haproxy_frontend_server.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021 Chris Morton, cosmo@cosmo.2y.net
```
Comment (opoplawski) [inline plugins/modules/pfsense_haproxy_frontend_server.py:4]: Might want to update the year.

File: plugins/modules/pfsense_haproxy_frontend.py (hunk 1 above)
Comment (cosmosified) [reply plugins/modules/pfsense_haproxy_frontend.py:17]: yep i made this over a year ago and never did a pr. client got back to me an dnow i'm having to get it to work again. i'll fix that.

File: plugins/modules/pfsense_haproxy_frontend_server.py (hunk 2 above)
Comment (cosmosified) [reply plugins/modules/pfsense_haproxy_frontend_server.py:4]: Thanks!

**Comment by cosmosified** [comment]
thats neat earlier wheni pusehd /created the pr it didn't run the builds. now i can see the failures

**Comment by opoplawski** [comment]
Thanks for the contribution.  However in addition to fixing the checks, I would want to see some unit tests as well for merging this.

**Comment by opoplawski** [comment]
> thats neat earlier wheni pusehd /created the pr it didn't run the builds. now i can see the failures

Yeah, I have to approve the test runs manually.

**Comment by cosmosified** [comment]
> Thanks for the contribution. However in addition to fixing the checks, I would want to see some unit tests as well for merging this.

yeah i figured as much. i pushed it up anyway for others to use. the tests came from actually implementing it. i'll see if can find time to follow how you have done tests and add some.

**Comment by opoplawski** [comment]
I would encourage you to create a topic branch for PRs instead of using master.

File: plugins/module_utils/__impl/interfaces.py [hunk 3]
```diff
@@ -140,3 +140,4 @@ def parse_interface(self, interface, fail=True, with_virtual=True):
     if fail:
         self.module.fail_json(msg='%s is not a valid interface' % (interface))
     return None
+
```
Comment (opoplawski) [inline plugins/module_utils/__impl/interfaces.py:143]: Drop the blank line, though I suspect the sanity test will flag this as well.

File: plugins/module_utils/pfsense.py [hunk 4]
```diff
@@ -582,6 +582,8 @@ def find_schedule_elt(self, name):
         """ return schedule elt if found """
         return self.find_elt_xpath("./schedules/schedule[name='{0}']".format(name))
 
+
+
```
Comment (opoplawski) [inline plugins/module_utils/pfsense.py:586]: This seems like a mistake

**Review by opoplawski — CHANGES_REQUESTED** [review]
Please fix the tests.  Thanks.

**Comment by opoplawski** [comment]
Also as a heads up, I'm likely going to move the haproxy stuff to it's own collections.

**Comment by cosmosified** [comment]
ok i haven' yhad a chance at all to look at what is required to get this merged. i will try this week. 
Will you move the backend ha-proxy things as well? is the idea to have seaprate collections for each package that could be installed/configured for pfsense??
thanks
cosmo

**Comment by cosmosified** [comment]
> I would encourage you to create a topic branch for PRs instead of using master.

hmm. i usually just create PRS from my local branch/feature branch to develop or main. are you suggesting that i do a pr from my fork branch to the saem branch on your fork? (ie the topic branch?) i ahven't done that before and apologize for the immediate ignorance on this. i will research this as well. thanks

**Comment by opoplawski** [comment]
> hmm. i usually just create PRS from my local branch/feature branch to develop or main. are you suggesting that i do a pr from my fork branch to the saem branch on your fork? (ie the topic branch?) i ahven't done that before and apologize for the immediate ignorance on this. i will research this as well. thanks

You filed a PR from the master branch of your fork.  You should create a new branch in your fork, say `haproxy` and make a PR from that.  That way you can work on multiple things at once.

**Comment by opoplawski** [comment]
> Will you move the backend ha-proxy things as well? is the idea to have seaprate collections for each package that could be installed/configured for pfsense??

Yup, that's the plan.

**Comment by cosmosified** [comment]
> > hmm. i usually just create PRS from my local branch/feature branch to develop or main. are you suggesting that i do a pr from my fork branch to the saem branch on your fork? (ie the topic branch?) i ahven't done that before and apologize for the immediate ignorance on this. i will research this as well. thanks
> 
> You filed a PR from the master branch of your fork. You should create a new branch in your fork, say `haproxy` and make a PR from that. That way you can work on multiple things at once.

yeah. i had it on a feature branch, merged that into my master, then requsted the pr back to yours. so my bad. thanks.

**Comment by opoplawski** [comment]
Okay, I've setup https://github.com/pfsensible/haproxy so please submit your PR there.  Thanks.

**Comment by opoplawski** [comment]
Moved to https://github.com/pfsensible/haproxy/pull/1

**Pull request closed without merging**